sandbox/libs/authn/src/main/java/org/opensearch/authn/User.java

```diff
@@ -25,8 +33,8 @@ public StringPrincipal getPrimaryPrincipal() {
     }
 
     @JsonProperty(value = "primary_principal")
-    public void setPrimaryPrincipal(StringPrincipal primaryPrincipal) {
-        this.primaryPrincipal = primaryPrincipal;
+    public void setPrimaryPrincipal(StringPrincipal principal) {
```

Comment:
Maybe swap the function to be just `setPrincipal` since you swap it from taking the `primaryPrincipal` argument.

Reply:
I'll revert this change to avoid confusion and follow the standard naming convention